pkg/minishift/util/fetch_iso_version.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type VersionSpec struct {
+	Assets []struct {
+		Name               string `json:"name"`
+		BrowserDownloadURL string `json:"browser_download_url"`
+	} `json:"assets"`
+}
+
+func GetLatestISOVersion(isoName string) string {
```

Comment:
In that case we will always have an issue in a way... cached images, in that case, should also be ignored and force a download of a compatible image. 

But what if I redownload an older version of Minishift, eg. v1.6.0, while the latest and the greatest is Minishift v2.0. How can I be sure the image that is compatible with v1.6.0 is used? this should in that case JUST be a single version... ? (Because I can just download a point release which was marked as compatible... or can I?)

I think we need to be more precise about how to version the ISOs in the first place, if that is the case.

But, I am of the opinion that nothing in the image should REALLY be specific to minishift. Minishift should do the downgrade or feature check.

Example. If I use Minishift which can support hostfolders using NFS, I need to check first if the image is compatible with this functionality (see if `mount.nfs` is available). I should not have to rely on the image...

and why? because if we target a VM (or cloud image) that is NOT provided or provisioned by us, we should also be able to start and work.

I can image that the pre-flight checks in the future can be extended to perform feature checks, and record this for use of Minishift

```
-- Check target for support features ...
   Check if VM support CIFS mounts ... OK
   Check if VM support SSHFS mounts ... OK
   Check if VM support NFS mounts ... FAIL
```